internal/agent/device/status/containers_test.go

```diff
@@ -144,10 +195,41 @@ var _ = Describe("containers exporter", func() {
 			Expect((*deviceStatus.Containers)[0].Image).To(Equal("quay.io/image1:latest"))
 			Expect((*deviceStatus.Containers)[0].Name).To(Equal("myfirstname"))
 			Expect((*deviceStatus.Containers)[0].Status).To(Equal("running"))
+			Expect((*deviceStatus.Containers)[0].Engine).To(Equal("podman"))
 			Expect((*deviceStatus.Containers)[1].Id).To(Equal("id2"))
 			Expect((*deviceStatus.Containers)[1].Image).To(Equal("quay.io/image2:latest"))
 			Expect((*deviceStatus.Containers)[1].Name).To(Equal("agreatname"))
 			Expect((*deviceStatus.Containers)[1].Status).To(Equal("paused"))
+			Expect((*deviceStatus.Containers)[1].Engine).To(Equal("podman"))
+
+			Expect(*deviceStatus.Conditions).To(HaveLen(1))
+			Expect((*deviceStatus.Conditions)[0].Type).To(Equal(v1alpha1.DeviceContainersRunning))
+			Expect((*deviceStatus.Conditions)[0].Status).To(Equal(v1alpha1.ConditionStatusFalse))
+			Expect(*(*deviceStatus.Conditions)[0].Reason).To(Equal("NotRunning"))
+			Expect(*(*deviceStatus.Conditions)[0].Message).To(Equal("1 container not running"))
+
+		})
+
+		It("list crio containers", func() {
+			container.matchPatterns = []string{"alpine", "busybox"}
+			execMock.EXPECT().LookPath("crictl").Return("/usr/bin/crictl", nil)
+			execMock.EXPECT().LookPath("podman").Return("", fmt.Errorf("not found"))
+			execMock.EXPECT().ExecuteWithContext(gomock.Any(), "/usr/bin/crictl", "ps", "-a", "--output", "json", "--name", "alpine", "--name", "busybox").Return(crioListResult, "", 0)
+			err := container.Export(context.TODO(), &deviceStatus)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(*deviceStatus.Containers).ToNot(BeNil())
+			Expect(len(*deviceStatus.Containers)).To(Equal(2))
+			Expect((*deviceStatus.Containers)[0].Id).To(Equal("id1"))
+			Expect((*deviceStatus.Containers)[0].Image).To(Equal("docker.io/library/alpine@sha256:c4a262d530f57d1b7b68b52ba8383c2e55fd1a0cb5b4f46b11eed7a2c4e143da"))
+			Expect((*deviceStatus.Containers)[0].Name).To(Equal("alpine"))
+			Expect((*deviceStatus.Containers)[0].Status).To(Equal("CONTAINER_RUNNING"))
+			Expect((*deviceStatus.Containers)[0].Engine).To(Equal("crio"))
+			Expect((*deviceStatus.Containers)[1].Id).To(Equal("id2"))
+			Expect((*deviceStatus.Containers)[1].Image).To(Equal("docker.io/library/busybox@sha256:4be429a5fbb2e71ae7958bfa558bc637cf3a61baf40a708cb8fff532b39e52d0"))
+			Expect((*deviceStatus.Containers)[1].Name).To(Equal("busybox"))
+			Expect((*deviceStatus.Containers)[1].Status).To(Equal("CONTAINER_EXITED"))
+			Expect((*deviceStatus.Containers)[1].Engine).To(Equal("crio"))
```

Comment:
I would add another test at least that runs with both podman and crio and verify that we have 4 containers.

This would handle if someone for example modifies your current code order (podman/crio...) where podman creates, and crio adds... which could get broken otherwise.

We can add it as a follow up too. For example we need to see how to get gomock to expect one call then another... same for lookpath.

Reply:
ok, I can do a followup PR with that. I need to learn a bit more about Ginkgo and Gomock to be honest :-)